projects/nx-verdaccio/src/plugin/targets/environment.targets.unit-test.ts

```diff
@@ -0,0 +1,463 @@
+    const result = isEnvProject(PROJECT_CONFIG, options);
+    expect(result).toBe(false);
+  });
+
+  it('should return true if targetNames match and tags are not present', (): void => {
+    const config = { ...PROJECT_CONFIG, tags: null };
+    const result = isEnvProject(config, OPTIONS);
+    expect(result).toBe(true);
+  });
+
+  it('should return true if targetNames match and filterByTags are not present', (): void => {
+    const options = {
+      ...OPTIONS,
+      filterByTags: null,
+    };
+    const result = isEnvProject(PROJECT_CONFIG, options);
+    expect(result).toBe(true);
+  });
+
+  it('should return true if targetNames match and tags match filterByTags', (): void => {
+    const result = isEnvProject(PROJECT_CONFIG, OPTIONS);
+    expect(result).toBe(true);
+  });
+
+  it('should return false if targetNames match but tags don’t match filterByTags', (): void => {
+    const options = {
+      ...OPTIONS,
+      filterByTags: ['mock-tag-no-match'],
+    };
+    const result = isEnvProject(PROJECT_CONFIG, options);
+    expect(result).toBe(false);
+  });
+});
+
+describe('verdaccioTargets', (): void => {
+  const port = 1337;
+  const customOption = 'custom-value';
+  const options = { ...OPTIONS, customOption }
+
+  let uniquePortSpy: MockInstance<[], number>;
+
+  vi.mock('node:path', (): { join: Mock } => {
+    return {
+      join: vi.fn().mockReturnValue('mocked-join'),
+    };
+  });
+
+  beforeEach((): void => {
+    uniquePortSpy = vi
+      .spyOn(uniquePortModule, 'uniquePort')
+      .mockReturnValue(port);
+  });
+  afterEach((): void => {
+    vi.clearAllMocks();
+    uniquePortSpy.mockRestore();
+  });
+
+  it('should generate verdaccio targets with correct structure', (): void => {
+    const result: Record<string, TargetConfiguration> = verdaccioTargets(
+      PROJECT_CONFIG,
+      options
+    );
+
+    expect(result).toMatchObject({
+      [TARGET_ENVIRONMENT_VERDACCIO_START]: expect.any(Object),
+      [TARGET_ENVIRONMENT_VERDACCIO_STOP]: expect.any(Object),
+    });
+  });
+
+  it('should generate verdaccio targets TARGET_ENVIRONMENT_VERDACCIO_START nested object with correct data ', (): void => {
+    const result: Record<string, TargetConfiguration> = verdaccioTargets(
+      PROJECT_CONFIG,
+      options
+    );
+
+    expect(result[TARGET_ENVIRONMENT_VERDACCIO_START]).toEqual({
+      outputs: [`{options.environmentRoot}/${VERDACCIO_STORAGE_DIR}`],
+      executor: '@nx/js:verdaccio',
+      options: {
+        config: '.verdaccio/config.yml',
+        port: port,
+        storage: JOIN_RESULT,
+        clear: true,
+        environmentDir: JOIN_RESULT,
+        projectName: PROJECT_NAME,
+        customOption: customOption,
+        filterByTags: TAGS,
+        targetNames: TARGET_NAMES,
+      },
+    });
+  })
+
+  it('should generate verdaccio targets TARGET_ENVIRONMENT_VERDACCIO_START nested object with correct data ', (): void => {
```

Comment:
```suggestion
  it('should generate the TARGET_ENVIRONMENT_VERDACCIO_START target correctly ', (): void => {
```
This seems a bit better phrased. You can apply it on multiple places in this file.